go-controller/pkg/ovn/namespace.go

```diff
@@ -330,26 +330,38 @@ func (oc *DefaultNetworkController) getAllHostNamespaceAddresses() []net.IP {
 				// skip hybrid overlay nodes
 				continue
 			}
-			hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node, types.DefaultNetworkName)
+			hostNetworkIPs, err := oc.getHostNamespaceAddressesForNode(node)
 			if err != nil {
-				klog.Warningf("Error parsing host subnet annotation for node %s (%v)",
-					node.Name, err)
-			}
-			for _, hostSubnet := range hostSubnets {
-				mgmtIfAddr := util.GetNodeManagementIfAddr(hostSubnet)
-				ips = append(ips, mgmtIfAddr.IP)
-			}
-			// for shared gateway mode we will use LRP IPs to SNAT host network traffic
-			// so add these to the address set.
-			lrpIPs, err := util.ParseNodeGatewayRouterLRPAddrs(node)
-			if err != nil {
-				klog.Errorf("Failed to get join switch port IP address for node %s: %v", node.Name, err)
-			}
-
-			for _, lrpIP := range lrpIPs {
-				ips = append(ips, lrpIP.IP)
+				klog.V(5).Infof("Error parsing annotation for node %s: %v", node.Name, err)
 			}
+			ips = append(ips, hostNetworkIPs...)
 		}
 	}
 	return ips
 }
+
+func (oc *DefaultNetworkController) getHostNamespaceAddressesForNode(node *kapi.Node) ([]net.IP, error) {
+	var ips []net.IP
+	hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node, types.DefaultNetworkName)
+	if err != nil {
+		klog.Warningf("Error parsing host subnet annotation for node %s: %v",
+			node.Name, err)
+		return nil, err
+	}
+	for _, hostSubnet := range hostSubnets {
+		mgmtIfAddr := util.GetNodeManagementIfAddr(hostSubnet)
+		ips = append(ips, mgmtIfAddr.IP)
+	}
+	// for shared gateway mode we will use LRP IPs to SNAT host network traffic
+	// so add these to the address set.
+	lrpIPs, err := util.ParseNodeGatewayRouterLRPAddrs(node)
+	if err != nil {
+		klog.Errorf("Failed to get join switch port IP address for node %s: %v", node.Name, err)
```

Comment:
same here, please explain why you are printing and returning the error here.

Reply:
I will stop printing Error message and just return the error if there is any. I need to return error from getHostNamespaceAddressesForNode to use it in add/delete IP to host network addr_set function.